app/controllers/products_controller.rb

```diff
@@ -3,7 +3,7 @@ def index
   end
 
   def show
-    @product = Product.find_by_id(params[:id])
+    @product = Product.find_by_id(params[:id]).decorate
```

Comment:
Fixed this issue and all other recommendations raised during the code review session.